tests/mainnet.rs

```diff
@@ -53,20 +57,30 @@ fn mainnet_blocks_from_rpc() {
             fs::create_dir_all(dir.clone()).unwrap();
             let file_block = File::create(format!("{dir}/block.json")).unwrap();
             serde_json::to_writer(file_block, &block).unwrap();
-            let file_state = File::create(format!("{dir}/state_for_execution.json")).unwrap();
-            serde_json::to_writer(file_state, &rpc_storage.get_cache()).unwrap();
+
+            let accounts: BTreeMap<Address, PlainAccount> =
+                rpc_storage.get_cache_accounts().into_iter().collect();
+            let file_state = File::create(format!("{dir}/pre_state.json")).unwrap();
+            serde_json::to_writer(file_state, &accounts).unwrap();
+
+            let block_hashes: BTreeMap<U256, B256> =
+                rpc_storage.get_cache_block_hashes().into_iter().collect();
+            if !block_hashes.is_empty() {
+                let file = File::create(format!("{dir}/block_hashes.json")).unwrap();
+                serde_json::to_writer(file, &block_hashes).unwrap();
+            }
         }
     }
 }
 
 #[test]
 fn mainnet_blocks_from_disk() {
-    common::for_each_block_from_disk(|block, state| {
+    common::for_each_block_from_disk(|block, state, block_hashes| {
```

Comment:
If it's not much of a hassle, renaming these `state`s as `pre_state`s will make it clearer.